ml-proto/host/builtins.ml

```diff
@@ -1,9 +1,50 @@
 open Source
+open Eval
+open Memory
+open Types
 
-let print vs =
+let print m vs =
   List.iter Print.print_value (List.map (fun v -> Some v) vs);
   None
 
+let rec stdout_write_inner at mem offset count i =
+  let load_result = Memory.load_extend mem (Int64.add offset i) Mem8 ZX Int32Type in
+
+  begin
+    match load_result with
+    | Values.Int32 byte ->
+      print_char (Char.chr (Int32.to_int byte))
+    | _ ->
+      ignore (Error.error at "load_extend returned wrong type")
+  end;
+
+  let next = Int64.succ i in
+  let should_iterate = (Int64.compare next count) < 0 in
+
+  if should_iterate then
+    ignore (stdout_write_inner at mem offset count next)
+  else
+    ();
```

Comment:
If I understand what's going on here, the `else ()` here is redundant; is that right, or am I missing something subtle?